src/components/structures/MessagePanel.tsx

```diff
@@ -1290,12 +1290,11 @@ class MainGrouper extends BaseGrouper {
         const keyEvent = this.events.find(e => this.panel.grouperKeyMap.get(e));
         const key = keyEvent ? this.panel.grouperKeyMap.get(keyEvent) : this.generateKey();
         if (!keyEvent) {
-            // Populate the weak map with the key that we are using for all related events.
-            this.events.forEach(e => {
-                if (!this.panel.grouperKeyMap.has(e)) {
-                    this.panel.grouperKeyMap.set(e, key);
-                }
-            });
+            // Populate the weak map with the key.
+            // Note that we only set the key on the specific event it refers to, since this group might get
+            // split up in the future by other intervening events. If we were to set the key on all events
```

Comment:
When would this split up happen?

Reply:
I wasn't able to find the specific cause, but I think what's happening is that when restarting Element, it pre-populates the timeline with state events that it knows about. Fetching the actual messages for the room can then split up these summaries.